### DISCUSSION TITLE
How to integrate filagui into project?

### DISCUSSION THREAD

**Discussion by noergel1** [google/filament#6651] [Q&A] [answered]
Hey,

I am trying to use filagui in my application, but I'm getting linking errors doing so.
I added the built filagui.lib to my VS project and tried instantiating an ImGuiHelper. The problem is that I keep getting 
"LNK2019	unresolved external symbol __imp_PathIsRelativeA referenced in function "public: bool __cdecl utils::Path::isAbsolute(void)const filament_sandbox	..\utils.lib(Path.cpp.obj)"
I tried linking both utils libraries (filament/out/install/lib/utils.lib and filament/out/libs/utils/RelWithDebInfo/utils.lib) and both don't fix it. 
It seems to be the case that the isAbsolute function doesn't get defined because Win32 is defined. I'm unsure how the libs are meant to be used, hope someone can help me out here.

Thanks in advance for your help! And sorry if the question is stupid, I'm not a very experienced programmer.

**Comment by bejado** [comment, marked as answer]
Just a guess, but you might need to adjust the `UNICODE` preprocessor definition: https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathisrelativea

**Reply by noergel1** [reply]
> Just a guess, but you might need to adjust the `UNICODE` preprocessor definition: https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathisrelativea

You're right, I also just found it. I had to include Shlwapi.lib. Thank you very much!
